lib/Agrammon/Web/Service.pm6

```diff
@@ -271,14 +272,27 @@ class Agrammon::Web::Service {
         $model-path.IO.&child-secure($technical).slurp
     }
 
-    method get-outputs-from-csv(
-        Agrammon::DB::User $user,
-        Str $simulation-name, Str $dataset-name, $csv-data,
+    subset InputFormats of Str where  { $_ eq 'application/json' or $_ eq 'text/csv' };
+    subset OutputFormats of Str where { $_ eq 'application/json' or $_ eq 'text/csv' or $_ eq 'text/plain' };
+
+    #| Run model from input data as CSV or JSON
+    #| and return output formatted as CSV, JSON, or TEXT
+    method get-outputs-for-rest(
+        Str $simulation-name, Str $dataset-name, $input-data, InputFormats $type,
         :$model-version, :$variants, :$technical-file,
-        :$language, :$format, :$print-only,
+        :$language, OutputFormats :$format!, :$print-only,
         :$include-filters, :$all-filters
     ) {
-        my $input = Agrammon::DataSource::CSV.new.from-csv($simulation-name, $dataset-name, $csv-data);
+        my $data-source;
+        given $type {
+            when 'text/csv' {
+                $data-source = Agrammon::DataSource::CSV.new;
+            }
+            when 'application/json' {
+                $data-source = Agrammon::DataSource::JSON.new;
```

Comment:
Note that instead of declaring a variable and then assigning in every branch, one can do `my $data-source = do given $type { when 'text/csv' { Agrammon::DataSource::CSV.new } ... }`.

Reply:
Changed.